packet/probe.c

```diff
@@ -323,7 +323,7 @@ int find_source_addr(
        anything to the port.
      */
     *sockaddr_port_offset(&dest_with_port) = htons(1);
-    len = sockaddr_addr_size(&dest_with_port);
+    len = sockaddr_size(&dest_with_port);
```

Comment:
`sockaddr_addr_size` returns size of `struct in_addr` or `struct in6_addr`, while `sockaddr_size` returns size of `struct sockaddr_in` or `struct sockaddr_in6`, `connect` expects the latter.